src/common/print.cpp

```diff
@@ -47,12 +49,12 @@ class PrintOutput {
 
 class SerialPrintOutput : public PrintOutput {
 public:
-  SerialPrintOutput() : PrintOutput(reinterpret_cast<char*>(UART_BASE)) {}
+  SerialPrintOutput() : PrintOutput(reinterpret_cast<char*>(UART_BASE)) {
+    platform_serial.init();
+  }
 
   void write(int chr) const final {
```

Comment:
Also I just noticed these "final", I forgot C++ even had this feature. Good change.